crates/notary/server/src/settings.rs

```diff
@@ -0,0 +1,77 @@
+use config::{Config, ConfigError, Environment, File};
+use std::path::Path;
+use tracing::{info, warn};
+use crate::{CliFields, NotaryServerProperties};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    #[serde(flatten)]
+    pub config: NotaryServerProperties,
+}
+
+impl Settings {
+    pub fn new(cli_fields: &CliFields) -> Result<Self, ConfigError> {
+        let mut builder = Config::builder();
+
+        // Add default values
+        builder = builder
+            .set_default("server.name", "notary-server")?
+            .set_default("server.host", "0.0.0.0")?
+            .set_default("server.port", 7047)?
+            .set_default("server.html-info", "<h1>Notary Server</h1>")?
+            .set_default("notarization.max-sent-data", 4096)?
+            .set_default("notarization.max-recv-data", 16384)?
+            .set_default("tls.enabled", true)?
+            .set_default("tls.private-key-pem-path", "../fixture/tls/notary.key")?
+            .set_default("tls.certificate-pem-path", "../fixture/tls/notary.crt")?
+            .set_default("notary-key.private-key-pem-path", "../fixture/notary/notary.key")?
+            .set_default("notary-key.public-key-pem-path", "../fixture/notary/notary.pub")?
+            .set_default("logging.level", "DEBUG")?
+            .set_default("logging.filter", Option::<String>::None)?
+            .set_default("authorization.enabled", false)?
+            .set_default("authorization.whitelist-csv-path", "../fixture/auth/whitelist.csv")?;
+
+        // Add the config file if it exists
+        let config_path = Path::new(&cli_fields.config_file);
+        if config_path.exists() {
+            info!("Loading configuration from: {}", cli_fields.config_file);
+            builder = builder.add_source(File::from(config_path));
+        } else {
+            warn!("Config file not found: {}. Using defaults and overrides.", cli_fields.config_file);
+        }
+
+        // Add environment variables
+        builder = builder.add_source(Environment::with_prefix("NOTARY_SERVER").separator("__"));

```

Comment:
question: so env var will override config file; and CLI will override both of them; is that right?

Reply:
@anthonykimani can u help to address this question?

Reply:
yes the CLI arguments will hold the highest priority, followed with the env var and config file last : )